airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -0,0 +1,486 @@
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if self.is_finished:
+            return None
+        return response.json().get("before_cursor")
+
+
+class IncrementalBasicSortedPageStream(IncrementalBasicUnsortedPageStream, ABC):
+    """basic stream for loading sorting data with normal pagination"""
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, next_page_token, **kwargs)
+        if params:
+            params.update({"sort_by": self.cursor_field, "sort_order": "desc", "limit": self.state_checkpoint_interval})
+        return params
+
+
+class TicketComments(IncrementalBasicSortedPageStream):
+    """TicketComments stream: https://developer.zendesk.com/api-reference/ticketing/tickets/ticket_comments/
+    ZenDesk doesn't provide API for loading of all comments by one direct endpoints.
+    Thus at first we loads all updated tickets and after this tries to load all created/updated
+    comments per every ticket"""
+
+    response_list_name = "comments"
+    cursor_field = IncrementalBasicSortedPageStream.created_at_field
+
+    class Tickets(IncrementalBasicSortedPageStream):
+        def request_params(
+            self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+        ) -> MutableMapping[str, Any]:
+            """Adds the field 'comment_count' for skipping tickets without comment"""
+            params = super().request_params(stream_state, next_page_token)
+            params["include"] = "comment_count"
+            return params
+
+    def path(self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        ticket_id = stream_slice["id"]
+        return f"tickets/{ticket_id}/comments.json"
+
+    def stream_slices(
+        self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        """Loads all updated tickets after last stream state"""
+        stream_state = stream_state or {}
+        tickets = self.Tickets(self._start_date, subdomain=self._subdomain, authenticator=self.authenticator).read_records(
+            sync_mode=sync_mode, cursor_field=cursor_field, stream_state={self.updated_at_field: stream_state.get(self.cursor_field)}
+        )
+        # selects all tickets what have at least one comment
+        stream_state = self.str2datetime(stream_state.get(self.cursor_field))
+        ticket_ids = [
+            {
+                "id": ticket["id"],
+                "start_stream_state": stream_state,
+            }
+            for ticket in tickets
+            if ticket["comment_count"]
+        ]
+        self.logger.info(f"Found updated tickets with comments: {[t['id'] for t in  ticket_ids]}")
```

Comment:
this log line could be huge, we can probably skip it

Reply:
Agree,  I've updated this message and it would print total  tickets' amount which were updated.